image/docker_schema1.go

```diff
@@ -112,6 +113,17 @@ func (m *manifestSchema1) ConfigBlob() ([]byte, error) {
 	return nil, nil
 }
 
+// Config returns the image configuration as per OCI v1 image-spec
+func (m *manifestSchema1) Config() (*imgspecv1.Image, error) {
```

Comment:
Yeah, computing DiffIDs would be extremely awkward.

But modifying `manifestSchema1.convertToManifestSchema2` to accept `nil` values for the `uploadedLayerInfos` and `layerDiffIDs` parameters, and to put invalid/zero/other placeholder values in the generated config instead of dereferencing the arrays if they are `nil`, seems plausible. Sure, a bit ugly, but better than having two entirely different code paths I think.